nova/virt/vmwareapi/driver.py

```diff
@@ -1039,12 +1056,19 @@ def post_live_migration_at_destination(self, context, instance,
                                            block_migration=False,
                                            block_device_info=None):
         """Post operation of live migration at destination host."""
-        self._vmops.disable_drs_if_needed(instance)
-        self._vmops.update_cluster_placement(context, instance)
-
-        volumes = self._get_volume_mappings(context, instance)
-        LOG.debug("Fixing shadow vms %s", volumes, instance=instance)
-        self._volumeops.fixup_shadow_vms(instance, volumes)
+        with self._error_out_instance_on_exception(instance,
+                "disable drs"):
+            self._vmops.disable_drs_if_needed(instance)
+
+        with self._error_out_instance_on_exception(instance,
+                "update cluster placement"):
+            self._vmops.update_cluster_placement(context, instance)
+
+        with self._error_out_instance_on_exception(instance,
+                "fixup shadow vms"):
+            volumes = self._get_volume_mappings(context, instance)
+            LOG.debug("Fixing shadow vms %s", volumes, instance=instance)
+            self._volumeops.fixup_shadow_vms(instance, volumes)
```

Comment:
If only one of those fails, we continue with the others - with an instance in error state. So we do as much as we can or as much as works at least. That's by design, right?

Reply:
Exactly. The order of those blocks is arbitrary and each item is kind of independent from one another. Personally, I prefer to get as much of it done as possible.